src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -0,0 +1,87 @@
+import com.google.gson.Gson;
+import com.google.sps.data.DatastoreScheduledInterviewDao;
+import com.google.sps.data.ScheduledInterview;
+import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/scheduled-interviews")
+public class ScheduledInterviewServlet extends HttpServlet {
+
+  private ScheduledInterviewDao scheduledInterviewDao;
+
+  @Override
+  public void init() {
+    init(new DatastoreScheduledInterviewDao());
+  }
+
+  // TODO: add a FakeScheduledInterviewDao class so this will become useful
+  public void init(ScheduledInterviewDao scheduledInterviewDao) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+  }
+
+  // Gets the current user's email from request and returns the ScheduledInterviews for that person.
+  // If the email that is requested matches the email that is logged in, then the scheduled
+  // interviews are returned, otherwise SC_UNAUTHORIZED is returned.
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String requestedEmail = request.getParameter("userEmail");
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    if (!requestedEmail.equals(userEmail)) {
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+    List<ScheduledInterview> scheduledInterviews =
+        scheduledInterviewDao.getForPerson(requestedEmail);
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(scheduledInterviews));
+  }
+
+  // Send the request's contents to Datastore in the form of a new ScheduledInterview object. If the
+  // email that is requested as either an interviewer or interviewee matches the email that is logged in, then the scheduled interview is
+  // stored, otherwise SC_UNAUTHORIZED is returned.
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String requestedInterviewerEmail = request.getParameter("interviewer");
+    String requestedIntervieweeEmail = request.getParameter("interviewee");
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    if (requestedInterviewerEmail.equals(userEmail)
+        || requestedIntervieweeEmail.equals(userEmail)) {
+      ScheduledInterview scheduledInterview =
+          ScheduledInterview.create(
+              -1,
+              new TimeRange(
+                  Instant.parse(request.getParameter("startTime")),
+                  Instant.parse(request.getParameter("endTime"))),
+              requestedInterviewerEmail,
+              requestedIntervieweeEmail);
```

Comment:
You should probably try-catch for Instant parsing errors around this call (check the specific exception `parse()` returns) and return a bad request error if parsing fails.

Reply:
Done